File: app/src/main/java/vergecurrency/vergewallet/service/model/wallet/TxProposal.java
package vergecurrency.vergewallet.service.model.wallet;

import com.google.gson.Gson;

public class TxProposal {

	private String address;
	private double amount;
	private String message;

	public static TxProposal decode(String JSON) {
		return new Gson().fromJson(JSON, TxProposal.class);
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public double getAmount() {
		return amount;
	}

	public void setAmount(double amount) {
		this.amount = amount;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/wallet/TxProposal.kt
package vergecurrency.vergewallet.service.model.wallet

import com.google.gson.Gson

class TxProposal {

    var address: String? = null
    var amount: Double = 0.toDouble()
    var message: String? = null

    companion object {

        fun decode(JSON: String): TxProposal {
            return Gson().fromJson(JSON, TxProposal::class.java)
        }
    }
}
